easybuild/easyblocks/s/siesta.py

```diff
@@ -0,0 +1,162 @@
+##
+# Copyright 2009-2016 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for Siesta, implemented as an easyblock
+
+@author: Miguel Dias Costa (National university of Singapore)
+"""
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.easyblocks.generic.makecp import MakeCp
+
+
+class EB_Siesta(MakeCp):
+    """Support for building and installing Siesta."""
+
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """
+        Define list of files or directories to be copied after make
+        """
+        extra = {
+            'files_to_copy': [[], "List of files or dirs to copy", CUSTOM],
+            'with_transiesta': [True, "Build transiesta", CUSTOM],
+            'with_utils': [True, "Build all utils", CUSTOM],
+        }
+        if extra_vars is None:
+            extra_vars = {}
+        extra.update(extra_vars)
+        return ConfigureMake.extra_options(extra_vars=extra)
```

Comment:
well, I meant to override, not extend, the `MakeCp.extra_options` (mainly because in `MakeCp`, `files_to_copy` is mandatory in the sense that it must be present in the easyconfig, and here I'm only using it in the easyblock)...